pandas/core/array_algos/putmask.py

```diff
@@ -0,0 +1,122 @@
+"""
+EA-compatible analogue to to np.putmask
+"""
+from typing import Any
+import warnings
+
+import numpy as np
+
+from pandas._libs import lib
+from pandas._typing import ArrayLike
+
+from pandas.core.dtypes.cast import convert_scalar_for_putitemlike, maybe_promote
+from pandas.core.dtypes.common import is_float_dtype, is_integer_dtype, is_list_like
+from pandas.core.dtypes.missing import isna_compat
+
+
+def putmask_simple(values: ArrayLike, mask: np.ndarray, value: Any):
```

Comment:
returns -> None

Reply:
though *maybe* we should return the array for consistency with putmask_simple?

Reply:
or rename to putmask_inplace

Reply:
> though maybe we should return the array for consistency with putmask_simple?

i think not returning anything makes it more obvious that it is inplace

> or rename to putmask_inplace

sure